packages/ui/v2/core/apps/AllApps.tsx

```diff
@@ -124,18 +128,13 @@ function CategoryTab({ selectedCategory, categories, searchText }: CategoryTabPr
   );
 }
 
-export default function AllApps({ apps, searchText }: AllAppsPropsType) {
+export default function AllApps({ apps, searchText, categories }: AllAppsPropsType) {
+  console.log("🚀 ~ file: AllApps.tsx:132 ~ AllApps ~ categories", categories);
   const router = useRouter();
   const { t } = useLocale();
   const [selectedCategory, setSelectedCategory] = useState<string | null>(null);
   const [appsContainerRef, enableAnimation] = useAutoAnimate<HTMLDivElement>();
 
-  const categories = apps
-    .map((app) => app.category)
-    .filter((cat, pos, self) => {
-      return self.indexOf(cat) === pos;
-    });
```

Comment:
By passing down categories as a prop, we don't need to loop through all the apps on the frontend.